core/src/main/java/io/lionweb/lioncore/java/serialization/PrimitiveValuesSerialization.java

```diff
@@ -3,15 +3,26 @@
 import com.google.gson.Gson;
 import com.google.gson.JsonElement;
 import com.google.gson.JsonParser;
+import io.lionweb.lioncore.java.metamodel.Enumeration;
+import io.lionweb.lioncore.java.metamodel.EnumerationLiteral;
 import io.lionweb.lioncore.java.metamodel.LionCoreBuiltins;
-import java.util.HashMap;
-import java.util.Map;
+import io.lionweb.lioncore.java.metamodel.Metamodel;
+import java.util.*;
 
 /**
  * This class is responsible for serialization and unserializing primitive values, based on the type
  * of the primitive value.
  */
 public class PrimitiveValuesSerialization {
+
+  private Map<String, Enumeration> enumerationsByID = new HashMap<>();
```

Comment:
`enumerationsByKey`?

Reply:
No, I think we want to use the ID for quickly and uniquely identify the enumerations (we still use the key to unserialize values)